### PR TITLE
Create project

### PR DESCRIPTION
**Pull request by JMurphyWeb**
# This almost completely unchanged since last review

# The addresses have been amended in pr #39 

# Ideally we will just merge in #39 and close this one as they have conflicting migration files

- [x] Run generator
- [x] fix up migration file
- [x] fix up model

#4

### CODE REVIEW

File: priv/repo/migrations/20170511142842_create_project.exs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      add :codename, :string
+      add :description, :text
+      add :start_date, :date
+      add :duration, :integer
+      add :studio_name, :string
+      add :company_name, :string
+      add :company_address_1, :string
+      add :company_address_2, :string
+      add :company_address_3, :string
+      add :company_address_4, :string
```
Comment (jackcarlisle) [inline priv/repo/migrations/20170511142842_create_project.exs]: did we want to change these to postcode, country etc?

File: test/controllers/project_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  test "renders form for new resources", %{conn: conn} do
+    conn = get conn, project_path(conn, :new)
+    assert html_response(conn, 200) =~ "New project"
+  end
+
+  test "creates resource and redirects when data is valid", %{conn: conn} do
+    conn = post conn, project_path(conn, :create), project: @valid_attrs
+    # assert redirected_to(conn) == project_path(conn, :index)
+    # assert Repo.get_by(Project, @valid_attrs)
```
Comment (jackcarlisle) [inline test/controllers/project_controller_test.exs]: should this still be commented out?

File: priv/repo/migrations/20170511142842_create_project.exs (hunk 1 above)
Comment (JMurphyWeb) [inline priv/repo/migrations/20170511142842_create_project.exs]: Yeah I was going to focus on the functionality/permissions/tests then come back to this.

**Comment by JMurphyWeb** [comment]
Thanks @jackcarlisle.

Let me add a bit more functionality then will look into splitting the PRs up

File: priv/repo/migrations/20170511142842_create_project.exs (hunk 1 above)
Comment (JMurphyWeb) [inline priv/repo/migrations/20170511142842_create_project.exs]: @jackcarlisle 

I just looked into changing the model to 
company_address_1
company_address_2
company_address_city
company_address_postcode
company_address_country

But I'm not sure if that is any better than 1, 2, 3, 4, 5 with the convention that they will be named appropriately to the user (5 = country, 4 = postcode, ...)

What are your thoughts?

Comment (jackcarlisle) [inline priv/repo/migrations/20170511142842_create_project.exs]: I guess it just makes it a bit easier for us, not having to remember which one correlates to each

Comment (JMurphyWeb) [inline priv/repo/migrations/20170511142842_create_project.exs]: done in #39

**Comment by jackcarlisle** [comment]
closing, changes absorbed by #39

**Pull request closed without merging**